### PR TITLE
Fix destructuring assignment of multiReturn (closes #995)

### LINKED ISSUE #995: Bug when destructuring a result directly.
I reproduce this bug using `string.find`, but maybe it applies to all `@tupleReturn`?

The following code reproduces this error on the tstl playground:
```
// works as it should
const result = string.find("a", "b");
const [c, ...d] = result;

// Doesn't work
const [a, ...b] = string.find("a", "b");
```

Gets translated to
```
result = {
    string.find("a", "b")
}
c = result[1]
d = __TS__ArraySlice(result, 1)
local ____ = string.find("a", "b")
a = ____[1]
b = __TS__ArraySlice(____, 1)
```

On the second case, it uses `string.find` result directly without putting it into a table, which causes a nil error.

### PR DESCRIPTION
**Pull request by yancouto**
Fixed the issue. Turns out we were dealing with this problem on normal assignments, so something like this works:
```js
let a, b;
[a, ...b] = string.find("a", "b");
```
This is handled here: https://github.com/TypeScriptToLua/TypeScriptToLua/blob/master/src/transformation/visitors/binary-expression/assignments.ts#L94-L97

Maybe this is an indicative that this needs some refactoring, as that code should be common. However, I just fixed the issue locally, and added a test showing it is now working.

The only other change is a snapshot of a test that's supposed to fail, so that shouldn't be a problem I believe.

`npm test` works.

### CODE REVIEW

File: test/unit/language-extensions/multi.spec.ts [hunk 1]
```diff
@@ -24,6 +24,19 @@ test("multi example use case", () => {
         .expectToEqual({ a: "foo", b: 5 });
 });
 
+test("multi #995", () => {
+    util.testModule`
+        function multiReturn(): LuaMultiReturn<[number, number, number]> {
+            return $multi(1, 2, 3);
```
Comment (Perryvw) [inline test/unit/language-extensions/multi.spec.ts:30]: Could you add another test where this function returns nil? (like the string.find in the issue)

File: test/unit/language-extensions/multi.spec.ts [hunk 2]
```diff
@@ -24,6 +24,19 @@ test("multi example use case", () => {
         .expectToEqual({ a: "foo", b: 5 });
 });
 
+test("multi #995", () => {
```
Comment (Perryvw) [inline test/unit/language-extensions/multi.spec.ts:27]: Our convention is to give the test a descriptive name, i.e `test("destructure LuaMultiReturn with rest"` and then add a comment with the PR link. (for example see https://github.com/TypeScriptToLua/TypeScriptToLua/blob/master/test/unit/builtins/object.spec.ts#L232 )

**Review by Perryvw — COMMENTED** [review]
Thanks for the PR, looks good! I only have some minor comments.

File: test/unit/language-extensions/multi.spec.ts [hunk 3]
```diff
@@ -24,6 +24,30 @@ test("multi example use case", () => {
...
+    util.testModule`
+        function multiReturn(): LuaMultiReturn<[number, number, number]> {
+            return;
+        }
+
+        const [a, ...b] = multiReturn();
+        export {a, b};
+    `
+        .setOptions(multiProjectOptions)
+        .expectToEqual({ b: [] });
```
Comment (Perryvw) [inline test/unit/language-extensions/multi.spec.ts:48]: Could you also check the value of `a` here? (should be undefined)

File: test/unit/language-extensions/multi.spec.ts [hunk 4]
```diff
@@ -24,6 +24,30 @@ test("multi example use case", () => {
...
+        function multiReturn(): LuaMultiReturn<[number, number, number]> {
+            return $multi(1, 2, 3);
+        }
+
+        const [a, ...b] = multiReturn();
+        export {a, b};
+    `
+        .setOptions(multiProjectOptions)
+        .expectToEqual({ a: 1, b: [2, 3] });
+    util.testModule`
```
Comment (Perryvw) [inline test/unit/language-extensions/multi.spec.ts:39]: Tests should only test one thing, could you split this second testModule into a second `test("Destructuring assignment of LuaMultiReturn returning nil", () => {`

**Review by Perryvw — APPROVED** [review]
